tests/src/test/java/org/wildfly/plugins/bootablejar/maven/goals/AbstractDevWatchTestCase.java

```diff
@@ -0,0 +1,220 @@
+                            cmd.add("pwsh.EXE");
+                            cmd.add("-command ");
+                            cmd.add("mvn");
+                            prop = "'" + prop + "'";
+                        } else {
+                            cmd.add("mvn.cmd");
+                        }
+                    } else {
+                        cmd.add("mvn");
+                    }
+
+                    String[] mvnCmd = {"-f", pomFile.toAbsolutePath().toString(), "wildfly-jar:dev-watch", "-e", prop};
+                    cmd.addAll(Arrays.asList(mvnCmd));
+                    logFile = getTestDir().resolve("target").resolve("dev-watch-test-output.txt");
+                    if (Files.exists(logFile)) {
+                        Files.delete(logFile);
+                    }
+                    final Path parent = logFile.getParent();
+                    if (parent != null && Files.notExists(parent)) {
+                        Files.createDirectories(parent);
+                    }
+                    process = new ProcessBuilder(cmd).redirectErrorStream(true)
+                            .redirectOutput(logFile.toFile()).start();
+                    int r = process.waitFor();
+                    if (r != 0) {
+                        retCode = r;
+                    }
+                } catch (Exception ex) {
+                    ex.printStackTrace();
+                    processException = ex;
+                }
+            }
+        };
+        goalThread = new Thread(r);
+        goalThread.start();
+        Exception waitException = null;
+        try (ModelControllerClient client = ModelControllerClient.Factory.create(TestEnvironment.getHost(), TestEnvironment.getManagementPort())) {
+            ServerHelper.waitForStandalone(client, TestEnvironment.getTimeout() * 5);
+            // Give some time for first deployment.
+            Thread.sleep(2000);
+        } catch (Exception ex) {
+            waitException = ex;
+        } finally {
+            if (waitException != null) {
+                if (processException != null) {
+                    waitException.addSuppressed(processException);
+                }
+                throw waitException;
+            }
+            if (processException != null) {
+                throw processException;
+            }
+        }
+    }
+
+    @Override
+    public void shutdownServer() throws Exception {
+        super.shutdownServer();
+
+        if (process != null) {
+            if (retCode != null) {
+                Exception ex = new Exception("dev-watch goal process not running although it should. Return code " + retCode);
+                if (processException != null) {
+                    ex.addSuppressed(processException);
+                }
+                throw ex;
+            }
+
+            Files.createFile(exitFile);
+            System.out.println("Created exit file, process should exit.");
+            goalThread.join();
```

Comment:
We may want to do something like `process.waitFor(timeout, TimeUnit.SECONDS)` here and possibly destroy forcefully destroy the process here in case some random error happens.

Reply:
Good point, will do.